samtranslator/intrinsics/actions.py

```diff
@@ -506,9 +500,9 @@ def resolve_resource_id_refs(
         remaining = splits[1:]  # if any
 
         resolved_value = supported_resource_id_refs.get(logical_id)
-        return self._get_resolved_dictionary(input_dict, key, resolved_value, remaining)  # type: ignore[no-untyped-call]
+        return self._get_resolved_dictionary(input_dict, key, resolved_value, remaining)
 
-    def _check_input_value(self, value):  # type: ignore[no-untyped-def]
+    def _check_input_value(self, value: Any) -> bool:
         # Value must be an array with *at least* two elements. If not, this is invalid GetAtt syntax. We just pass along
         # the input to CFN for it to do the "official" validation.
         if not isinstance(value, list) or len(value) < 2:
```

Comment:
It seems like we can use here `sam_expect` like in this example https://github.com/aws/serverless-application-model/blob/09fcb574c3dbf2ab5e9ab30651fae21a29c91fed/samtranslator/model/apigateway.py#L182

Reply:
I think `sam_expect` is too strong here. It will throw an exception if type doesn't match. However, we do not want to throw an exception here.